### ISSUE TITLE
Does not work for more advanced optimizers

### ISSUE THREAD

**Issue by Svito-zar** [cmgreen210/TensorFlowDeepAutoencoder#4]
Your code is working only with Gradient Descent Optimizer, which is by far not the best one.
If you try to change it to Adam,for example, you will have a lot of problems because of the system architecture. It is actually quit tricky to train a deep system layer-wise in tensorflow.

I am working on making it. Let me know, if you are interested in that.

**Comment by SawyerW** [comment]
@Svito-zar Hi，I am also trying to study the model，could you share it when you finish your work on it？ Thank you.

**Comment by Svito-zar** [comment]
Yes, sure and it is open sourse, so you can have a look on a current state anytime : https://github.com/Svito-zar/DAE-for-Mocap-Representation-Learning

By the way, one paper (https://arxiv.org/abs/1405.1380) is stating that learning jointly is better that layer-wise training and it was very hard to implement layer-wise training in Tensorflow properly, so I am training the whole network jointly now.

P.S. : Using Adam optimizer, instead of Gradient Descent improved my results more than in 2 times!

**Comment by brent-halen** [comment]
Svito-zar, your link is dead. Did it not work out after all?

**Comment by Svito-zar** [comment]
Now it is fine.
New version of Tensorflow solved an issue.
